docs/how-to/h-enable-monitoring.md

```diff
@@ -0,0 +1,215 @@
+prometheus:grafana-dashboard                  grafana:grafana-dashboard                        grafana_dashboard          regular  
+prometheus:grafana-source                     grafana:grafana-source                           grafana_datasource         regular  
+prometheus:prometheus-peers                   prometheus:prometheus-peers                      prometheus_peers           peer     
+scrape-interval-config:metrics-endpoint       prometheus:metrics-endpoint                      prometheus_scrape          regular  
+traefik:ingress                               alertmanager:ingress                             ingress                    regular  
+traefik:ingress                               catalogue:ingress                                ingress                    regular  
+traefik:ingress-per-unit                      loki:ingress                                     ingress_per_unit           regular  
+traefik:ingress-per-unit                      prometheus:ingress                               ingress_per_unit           regular  
+traefik:metrics-endpoint                      prometheus:metrics-endpoint                      prometheus_scrape          regular  
+traefik:traefik-route                         grafana:ingress                                  traefik_route              regular  
+
+
+```
+
+
+In order to send metrics to the Prometheus Pushgateway we need to get the address of the Prometheus PushGateway: 
+
+```shell
+export PROMETHEUS_GATEWAY=$(juju status --format=yaml | yq ".applications.prometheus-pushgateway-k8s.address") 
+export PROMETHEUS_PORT=9091
+```
+
+To enable the push of metrics you only need to add the following lines as configuration to a `spark-client` configuration file (e.g., `spark-defaults.conf`): 
+
+```shell
+spark.metrics.conf.driver.sink.prometheus.pushgateway-address=$PROMETHEUS_GATEWAY:$PROMETHEUS_PORT
+spark.metrics.conf.driver.sink.prometheus.class=org.apache.spark.banzaicloud.metrics.sink.PrometheusSink
+spark.metrics.conf.driver.sink.prometheus.enable-dropwizard-collector=true
+spark.metrics.conf.driver.sink.prometheus.period=5
+spark.metrics.conf.driver.sink.prometheus.metrics-name-capture-regex=([a-z0-9]*_[a-z0-9]*_[a-z0-9]*_)(.+)
+spark.metrics.conf.driver.sink.prometheus.metrics-name-replacement=\$2
+spark.metrics.conf.executor.sink.prometheus.pushgateway-address=$PROMETHEUS_GATEWAY:$PROMETHEUS_PORT
+spark.metrics.conf.executor.sink.prometheus.class=org.apache.spark.banzaicloud.metrics.sink.PrometheusSink
+spark.metrics.conf.executor.sink.prometheus.enable-dropwizard-collector=true
+spark.metrics.conf.executor.sink.prometheus.period=5
+spark.metrics.conf.executor.sink.prometheus.metrics-name-capture-regex=([a-z0-9]*_[a-z0-9]*_[a-z0-9]*_)(.+)
+spark.metrics.conf.executor.sink.prometheus.metrics-name-replacement=\$2
+```
+
+or as alternative you can directly add each option as configuration option to `spark-client.spark-submit`: 
+
+```shell
```

Comment:
I would skip this, and I would just tell user that they can also feed these arguments directly to the spark-submit command, as shown [here](https://discourse.charmhub.io/t/spark-client-snap-tutorial-spark-submit/8953).